plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/generator/SQLGeneratorDialog.java

```diff
@@ -85,6 +86,9 @@ protected Composite createDialogArea(Composite parent) {
                 if (supportPermissions && supportComments && supportFullDDL) {
                     break; //it supports everything
                 }
+                if (sourceObject.supportsObjectDefinitionOption(DBPScriptObject.OPTION_INCLUDE_PARTITIONS)) {
+                    supportsPartitionsDDL = true;
+                }
```

Comment:
supportsPartitionsDDL = sourceObject.supportsObjectDefinitionOption(DBPScriptObject.OPTION_INCLUDE_PARTITIONS)

Reply:
Well, I see that all code in this method is written this way. Okay